src/python/foglamp/core/scheduler.py

```diff
@@ -6,50 +6,349 @@
 
 """FogLAMP Scheduler"""
 
+import time
+import datetime
+from enum import IntEnum
 import asyncio
-from asyncio.subprocess import Process
+import logging
+import collections
+import uuid
+import sys # TODO: Needed for logging delete me
+
+import aiopg.sa
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql as pg_types
 
 __author__ = "Terris Linenbach"
 __copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
-# For Process methods, see https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process
-# or upgrade to a version of Python that uses type annotations
-_processes = []  # type: List[Process]
 
+class Scheduler(object):
```

Comment:
``` 
  scheduler = Scheduler() 
   # call it TaskScheduler? Scheduler is fine too.. 
   # consider to have BaseScheduler and subclass it?

   # Move Db stuff to TaskSchedulerStore or task_scheduler_store.py ?

   scheduler.add_task(task0, _ScheduleType.<TIMED/ INTERVAL>, time=?, interval=?) 

   # scheduler.remove_task(task0)

   # I see that we are processing db objects array for it.. but read db and add to scheduler.. let's process in this way? Atleast this will be the way for REST API to tell db that I want this task to be get started/ stopped... individual task  and then this code will process those db objects 

   # task0 is instance/ def of Task
   # _ScheduleType.TIMED will need kwarg time else interval (with unit?), defaults to time?

    `# For "__main__"`
    scheduler.start()
    print('Press Ctrl+C to exit')

    try:
        # process tasks

    except (KeyboardInterrupt, SystemExit):
        # kill the instance




core server, cls.__scheduler.start()
it reads from storage, and do the stuff for tasks

but..

__scheduler.start only once and read storage once? or it should periodically read storage?
````